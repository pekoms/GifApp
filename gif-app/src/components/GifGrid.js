import { useEffect } from "react"
import { useState } from "react"
import {React} from 'react'
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";


export default function GifGrid({category}) {

      const {data:images,loading} = useFetchGifs(category);
    
    return (
        <>
        <h3 className="animate__fadeInRightBig">{category}</h3>
        {loading && <p className="animate__flash">Loading</p>}
        <div className="card-grid animate__fadeInLeft">
            
            
            {
                images.map(img=>(
                    <GifGridItem key={img.id}{...img}></GifGridItem>
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes={
    category: PropTypes.string.isRequired
}