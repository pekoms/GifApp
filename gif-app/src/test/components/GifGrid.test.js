import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');



describe('Match ',()=>{


    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs',()=>{
        const gifs=[{
            id:'ABC',
            url:'http://localhost/cualquier/cosa.jpg',
            title:'Cualquier cosa'
        },
        {
            id:'123',
            url:'http://localhost/cualquier/cosa.jpg',
            title:'Cualquier cosa'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });
        const category = 'Pokemon';
        const wrapper=shallow(<GifGrid category={category}/>)
        
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('p').exists()).toBe(false);
       expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
})