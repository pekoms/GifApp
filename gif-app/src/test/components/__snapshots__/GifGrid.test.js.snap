// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match  Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__fadeInRightBig"
  >
    Pokemon
  </h3>
  <div
    className="card-grid animate__fadeInLeft"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
